server/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java

```diff
@@ -58,10 +64,23 @@ public SysSnapshots(RepositoriesService repositoriesService) {
     }
 
     public Iterable<SysSnapshot> currentSnapshots() {
-        return () -> getRepositories.get().stream()
-            .flatMap(repository -> repository.getRepositoryData().getSnapshotIds().stream()
-                .map(snapshotId -> createSysSnapshot(repository, snapshotId))
-            ).iterator();
+        ArrayList<SysSnapshot> result = new ArrayList<>();
```

Comment:
This call will move from lazy to eager and it is not great because there is io involved.

Reply:
`currentSnapshots` could be changed to return a future of `Iterable<SysSnapshot>` see how it is used: 

https://github.com/crate/crate/blob/c5d71a4b87161314685c29d8901e6e56eebbffb0/server/src/main/java/io/crate/metadata/sys/SysTableDefinitions.java#L107-L110

Together with `CompletableFutures.allAsList` and `FutureActionListener` it should be possible to generate the whole result asynchronous